### DISCUSSION TITLE
CAD files that cannot be read

### DISCUSSION THREAD

**Discussion by Lukeor** [DomCR/ACadSharp#316] [Q&A]
Hey,

My line of work involves me and my colleagues regularly working with .dxf and .dwg files. We often come across a variety of CAD files that have been created in a wide variety of applications. Due to this, some of these files may have slight variations between each other and this is known to sometimes cause issues when viewing them in CAD viewers such as QGIS, AutoCAD etc.

ACadSharp is very good at reading in the vast majority of .dxf and .dwg files without any issues, but sometimes it can throw an exception when reading in a file.

I understand that the cause for these .dxf .dwg not being able to be read into ACadSharp is probably down to the validity of the file itself and the data within; I thought however it might be useful to share these problematic files with you in case you were interested in having a look and investigating the reason why. 

Thanks again for a great library.

[Problematic CAD Files.zip](https://github.com/DomCR/ACadSharp/files/14667065/Problematic.CAD.Files.zip)

Luke

**Comment by DomCR** [comment]
Hi @Lukeor,

I've checked the files, the dxf is widely incomplete, there is not even a version number but I'm guessing that is only meant to store some geometry, this branch is meant to make the reader more flexible due the differences between the dxf versions and other generators, I'll try to fix it.

- https://github.com/DomCR/ACadSharp/pull/165

The dwg seems to trigger a bug in `ACadSharp`, I'll open an issue to address it.

Thanks for your support!

**Reply by Lukeor** [reply]
Thank you as always for your prompt replies and willingness to investigate these issues.

They are very much appreciated.

All the best,
Luke

**Reply by Lukeor** [reply]
Hey again :) 

I've come across another strange DXF file that I have attached.
With this one, when viewing it in TrueView I get the following entities displayed:

![image](https://github.com/DomCR/ACadSharp/assets/162500307/e70cd07a-b799-4c92-b938-e1739a652cf8)

The file seems to get successfully read by ACadSharp, but when I have a look within the ACadSharp's file Block Records (including the Model Space one) I can't seem to find any entities that have been read in.

I suspect another incomplete CAD file, perhaps?

Thanks for your continued support!

Luke

[Problematic DXF.zip](https://github.com/DomCR/ACadSharp/files/14854051/Problematic.DXF.zip)

**Reply by DomCR** [reply]
Hi @Lukeor,

I've tested the file, it's a similar case, in this case the file does not have the owner for the entities assign so `ACadSharp` doesn't know where to put them, also there are some missing table entries that should be there like line types.

I'll open an issue to fix this one.

Thanks for reporting it.

**Reply by DomCR** [reply]
I've added this issue for a new feature to help with this kind of files:

- https://github.com/DomCR/ACadSharp/issues/339

**Reply by Lukeor** [reply]
Hi @DomCR,

Thanks for investigating the issue and for putting forward the new feature. It is much appreciated.

All the best.
